### DISCUSSION TITLE
How to get Helix to use a python virtual environment?

### DISCUSSION THREAD

**Discussion by clue44** [helix-editor/helix#4722] [Q&A]
When opening a python file, Helix marks the python imports with a red dot and says "[import-error] Unable to import 'requests'" for example.

It does this no matter if the virtual python environment is activated or not before opening Helix with the python file.

How do I get Helix to use a virtual environment in the same folder as the python code?

![image](https://user-images.githubusercontent.com/42569344/201470000-3c0a07b3-ea72-441b-a87a-b350f5e04342.png)

**Comment by archseer** [comment]
Helix just runs the language server as is. You need to configure the LSP (pylsp?) to pick up the env

**Comment by JorisL** [comment]
What I do is:

1. Install pylsp (and black) within the virtual environment.
2. Open a terminal and launch the virtual environment (`source myvenv/bin/activate` or `venv\Scripts\Activate.ps1`).
3. Launch helix from that terminal session.

Helix just launches pylsp without any specified path, so when the virtual environment is activated it will launch the pylsp from that environment (similar to directly running pylsp in the terminal after the environment is activated).

**Reply by goof312** [reply]
Is it ok to run venv in two tabs of window? One tab for helix and the other one is to run my django or flask?

**Reply by a3lem** [reply]
> Is it ok to run venv in two tabs of window? One tab for helix and the other one is to run my django or flask?

As in, is it okay to activate the same venv in multiple shell sessions? Yup, that's absolutely fine.

**Comment by malik-n** [comment]
Which language server are you using?
I am currently trying to [solve this for pyright](https://github.com/microsoft/pyright/issues/9796)

**Reply by MrRobotMan** [reply]
The only way I've found that works is to create a pyrightconfig.json.
```
{
  "venv": ".venv", 
  "venvPath": "." 
}
```

**Comment by goof312** [comment]
I use Windows Terminal.
